lib/vehicle_factory.rb

```diff
@@ -0,0 +1,25 @@
+class VehicleFactory
+  attr_accessor :registration_date
+  attr_reader   :vehicles
+
+  def initialize
+    @vehicles = []
+  end
+  
+  def create_vehicles(registrations)
+    registrations.map do |vehicle|
+      @vehicles << vehicle = Vehicle.new(registrations)
```

Comment:
Looks like you were in the process of building this out. Nice use of `.map`. So you probably won't want to create the `vehicle` variable here. You can instead just shovel `Vehicle.new` into your `@vehicles` array. 
So if you were to look at your `vehicle_spec.rb` you will see that when a new `vehicle` object is created, it does indeed only take 1 argument but that argument is a hash. So you will want to replicate that in this method. So something like so:

```
registrations.map do |vehicle|
    @vehicles << Vehicle.new({
        engine: all_electric,
        make: vehicle[:make],
        model: vehicle[:model],
        vin: vehicle[:vin_1_10],
        year: vehicle[:model_year]
    })
```